File: 2026-08/src/day_31/prms_181883/Solution2.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    int n = arr.size();
    vector<int> diff(n + 1);

    for (auto& q : queries) {
        diff[q[0]]++;
        diff[q[1] + 1]--;
    }

    int sum = 0;
    for (int i = 0; i < n; i++) {
        sum += diff[i];
        arr[i] += sum;
    }

    return arr;
}

File: 2026-08/src/day_31/prms_181883/Solution2.py
from itertools import accumulate


def solution(arr, queries):
    diff = [0] * (len(arr) + 1)
    for s, e in queries:
        diff[s] += 1
        diff[e + 1] -= 1

    return [x + d for x, d in zip(arr, accumulate(diff))]
